smithy-model/src/main/java/software/amazon/smithy/model/traits/ephemeral/OriginalShapeIdTrait.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.traits.ephemeral;
+
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.AbstractTrait;
+
+/**
+ * Used to provide the original shape ID of a shape that is renamed
+ * in the semantic model.
+ *
+ * <p>This is an ephemeral trait that is not defined in the semantic model,
+ * nor is it persisted when the model is serialized.
+ */
+public final class OriginalShapeIdTrait extends AbstractTrait {
```

Comment:
nit: this could be a `StringTrait`

Reply:
I didn’t use a StringTrait because I didn’t think the getValue method should return a string here as this trait is only meant for defining the previous shape ID. Fair?